language/move-prover/src/prelude.bpl

```diff
@@ -519,6 +519,27 @@ procedure {:inline 1} BorrowField(src: Reference, f: FieldName) returns (dst: Re
     dst := Reference(l#Reference(src), p);
 }
 
+procedure {:inline 1} GetGlobal(address: Value, ta: TypeValue) returns (dst: Value)
+{
+    var r: Reference;
+
+    call r := BorrowGlobal(address, ta);
```

Comment:
Specific reason why we needed new instructions for this? Seems to be just a combination of two existing ones. Same for GetFieldFromReference.

Reply:
I have added two "virtual" instructions in our bytecode IR---GetField and GetGlobal.  It is true that these instructions can be compiled down to a combination of two instructions, one that takes a reference and the other that reads the reference. But I am trying to avoid that creation of extra reference in the bytecode that will be translated to Boogie so that all immutable references at the *bytecode* level are eliminated.